api/reference/custom_actions.adoc

```diff
@@ -0,0 +1,383 @@
+
+When custom buttons are defined on a VM, the returned VM resource is extended with custom actions which
+can be triggered via a POST action.
+
+The action name to trigger the action is the lower-cased name of the custom button.
+
+As an example, here's a VM without custom buttons:
+
+[source,json]
+----
+{
+  "href": "https://172.16.30.129/api/vms/172",
+  "id": "172",
+  "vendor": "vmware",
+  "name": "aab-idp",
+  "location": "aab-idp/aab-idp.vmx",
+  "host_id": "4",
+  "created_on": "2018-01-09T19:23:37Z",
+  "updated_on": "2018-01-09T19:23:37Z",
+  "storage_id": "8",
+  ...
+  "actions": [
+    {
+      "name": "edit",
+      "method": "post",
+      "href": "https://172.16.30.129/api/vms/173"
+    },
+    {
+      "name": "add_lifecycle_event",
+      "method": "post",
+      "href": "https://172.16.30.129/api/vms/173"
+    },
+    {
+      "name": "add_event",
+      "method": "post",
+      "href": "https://172.16.30.129/api/vms/173"
+    },
+    ...
+    {
+      "name": "set_miq_server",
+      "method": "post",
+      "href": "https://172.16.30.129/api/vms/173"
+    },
+    {
+      "name": "scan",
+      "method": "post",
+      "href": "https://172.16.30.129/api/vms/173"
+    },
+    {
+      "name": "delete",
+      "method": "delete",
+      "href": "https://172.16.30.129/api/vms/173"
+    }
+  ]
+}
+----
+
+Here is an example of a VM with custom buttons:
+
+[source,json]
+----
+{
+  "href": "https://172.16.30.129/api/vms/173",
+  "id": "173",
+  "vendor": "vmware",
+  "name": "aab-ldap",
+  "location": "aab-ldap/aab-ldap.vmx",
+  "host_id": "4",
+  "created_on": "2018-01-09T20:23:37Z",
+  "updated_on": "2018-01-09T20:23:37Z",
+  "storage_id": "8",
+  "guid": "dee6d969-7c69-44b6-bcd3-3b14f21702c5",
+  "ems_id": "1",
+  "uid_ems": "420c8f0d-4a8f-0b6d-fc3a-0c3cfab7b33f",
+  "tools_status": "toolsNotRunning",
+  "standby_action": "checkpoint",
+  "power_state": "off",
+  ...
+  "actions": [
+    {
+      "name": "edit",
+      "method": "post",
+      "href": "https://172.16.30.129/api/vms/173"
+    },
+    {
+      "name": "add_lifecycle_event",
+      "method": "post",
+      "href": "https://172.16.30.129/api/vms/173"
+    },
+    {
+      "name": "add_event",
+      "method": "post",
+      "href": "https://172.16.30.129/api/vms/173"
+    },
+    ...
+    {
+      "name": "set_miq_server",
+      "method": "post",
+      "href": "https://172.16.30.129/api/vms/173"
+    },
+    {
+      "name": "scan",
+      "method": "post",
+      "href": "https://172.16.30.129/api/vms/173"
+    },
+    {
+      "name": "delete",
+      "method": "delete",
+      "href": "https://172.16.30.129/api/vms/173"
+    },
+    {
+      "name": "analyze",
+      "method": "post",
+      "href": "https://172.16.30.129/api/vms/173"
+    },
+    {
+      "name": "ping",
+      "method": "post",
+      "href": "https://172.16.30.129/api/vms/173"
+    }
+  ]
+}
+----
+
+Here, the _analyze_ and _ping_ are custom buttons for the VM and are returned after
```

Comment:
Actually, since you specify which ones are custom buttons here, we can probably disregard my above comment.